docs/decisions/2024-07-10_include_file_limits.md

```diff
@@ -0,0 +1,17 @@
+# 2024-03-26 Appropriate limit for include files per project
+
+- status: in review (will change to accepted after PR is approved)
+- date: 2024-07-11
+- authors: Bynn Lee
+
+## Context and Problem Statement
+
+The introduction of the `include` capability to project configuration has increased Evergreen's use of git clones since we ask GitHub for each individual included file separately. We needed a way to stop users from infinitely scaling the number of includes until we hit our GitHub API limit. 
+
+Include files take up a significant percentage of all of Evergreen's GitHub operations. After switching to GitHub apps, our new hourly limit is 15,000 requests to GitHub per hour. On average, we only use around 4-8000 of our limit per hour, depending on how active users are. Of those requests, around 60% can be contributed to retrieving include files. For example, on an hour we used 7000 of our limit, we would also have around 4000 include file retrieval requests logged. 
+
+The two biggest loads are the the mongo repo projects that have around 25 includes on average, and the mms repo projects which currently have 32 include files. However, the mongo projects have not added extra include files in the past year whereas the mms projects have added almost 10 new include files.
+
+## Decision Outcome
+
+I propose a limit of 40 includes on all projects. Assuming our GitHub usage scales consistently with what we have been seeing, we should be able to handle a 100% increase on the number of include files. So in theory, a limit of 50 should not exceed our limits. However, there are intermittent cases (about once a month) where we do end up using 10,000 of our API limit for the hour and setting the limit too high could use up our limit during those times. 
```

Comment:
Yes, if the limit is reached, it would make it very risky for us to exceed our hourly limit. I do think that 35 is a much safer limit (or even 30) and there could be room for people pushing back. 

I think it would be really important to communicate this limit to people (continuing this conversation in the other thread)

Reply:
30 sounds like a good small bump to start off, which we can monitor and then raise later if we think it's reasonable!

Reply:
The problem with 30 is that we would have to ask mms to get rid of 2 because they already use 32 😢

Reply:
Hmm, I guess 35 is okay since it's only 3 more then? We can keep an eye on it, but just giving a small bump over the current max seems good enough.